pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -102,24 +104,30 @@ func SetDomainSpec(virConn cli.Connection, vmi *v1.VirtualMachineInstance, wante
 	return dom, nil
 }
 
-func GetDomainSpec(dom cli.VirDomain) (*api.DomainSpec, error) {
-	spec, err := GetDomainSpecWithFlags(dom, libvirt.DOMAIN_XML_MIGRATABLE)
-	if err != nil {
-		return nil, err
-	}
+func GetDomainSpec(status libvirt.DomainState, dom cli.VirDomain) (*api.DomainSpec, error) {
 
+	var spec *api.DomainSpec
 	inactiveSpec, err := GetDomainSpecWithFlags(dom, libvirt.DOMAIN_XML_INACTIVE)
 	if err != nil {
 		return nil, err
 	}
 
+	spec = inactiveSpec
+	// libvirt (the whole server) sometimes block indefinitely if a guest-shutdown was performed
```

Comment:
Does this mean, that guest shutdown will cause endless wait for xml?
If so, this seems to fix the issue I have observed here: #1532 
Good job finding it!

Reply:
Yes, pretty strange.